samples/basic/blinky_pwm/boards/frdm_mcxc242.overlay

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&tpm1 {
```

Comment:
Any reason not to set these clocks in the base board devicetree?

Reply:
Unfortunately, common clock source cannot be used for all PWM samples and tests with TPM. I tested:
- samples\basic\blinky_pwm
- samples\basic\fade_led
- samples\basic\rgb_led
- samples\drivers\led\pwm
- tests\drivers\pwm\pwm_api

pwm_api test requires high frequency clock source. It PASSED only with 48MHz MCGPCLK. Potential usage of lower freq is impacted by limitations of TPM driver: fixed pre-scaler and inability to set frequency lower than 1 Hz.
On the other hand, some other samples like samples\drivers\led\pwm requires lower freq because it sets 0.2 sec period.